apps/AppWithWearable/lib/pages/memories/page.dart

```diff
@@ -176,13 +157,39 @@ class _MemoriesPageState extends State<MemoriesPage> with AutomaticKeepAliveClie
             : SliverList(
                 delegate: SliverChildBuilderDelegate(
                   (context, index) {
-                    return MemoryListItem(
-                      memoryIdx: index,
-                      memory: memories[index],
-                      loadMemories: widget.refreshMemories,
-                    );
+                    int dateIndex = 0;
```

Comment:
For some reason I think this implementation has very little performant.

We are going through memories like twice on every UI render flutter does, can we remove some of the `memories.where` from the build method?

Reply:
I'll do some testing with the implementation to make it more performant.